pkg/querier/tenantfederation/merge_queryable_test.go

```diff
@@ -195,241 +213,465 @@ func (m mockTenantQuerier) LabelNames() ([]string, storage.Warnings, error) {
-				actLabelValues, warnings, err := querier.LabelValues(labelName)
-				if tc.expectedQueryErr != nil {
-					require.EqualError(t, err, tc.expectedQueryErr.Error())
+			t.Run(scenario.labelNamesTestCase.name, func(t *testing.T) {
+				labelNames, warnings, err := querier.LabelNames()
+				if scenario.labelNamesTestCase.expectedQueryErr != nil {
+					require.EqualError(t, err, scenario.labelNamesTestCase.expectedQueryErr.Error())
 				} else {
 					require.NoError(t, err)
-					assert.Equal(t, expectedLabelValues, actLabelValues, fmt.Sprintf("unexpected values for label '%s'", labelName))
-					assertEqualWarnings(t, tc.expectedWarnings, warnings)
+					assert.Equal(t, scenario.labelNamesTestCase.expectedLabelNames, labelNames)
+					assertEqualWarnings(t, scenario.labelNamesTestCase.expectedWarnings, warnings)
 				}
+			})
+		})
+	}
+}
+
+func TestMergeQueryable_LabelValues(t *testing.T) {
+	// set a multi tenant resolver
+	tenant.WithDefaultResolver(tenant.NewMultiResolver())
+
+	for _, scenario := range []labelValuesScenario{
+		{
+			mergeQueryableScenario: singleTenantScenario,
+			labelValuesTestCases: []labelValuesTestCase{
+				{
+					name:                "should return all label values for instance when no matchers are provided",
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a"},
+				},
+				{
+					name:                "should return no tenant values for the __tenant_id__ label as the MergeQueryable has been bypassed",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: nil,
+				},
+			},
+		},
+		{
+			mergeQueryableScenario: singleTenantNoBypassScenario,
+			labelValuesTestCases: []labelValuesTestCase{
+				{
+					name:                "should return all label values for instance when no matchers are provided",
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a"},
+				},
+				{
+					name:                "should return a tenant team value for the __tenant_id__ label as the MergeQueryable has not been bypassed",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: []string{"team-a"},
+				},
+			},
+		},
+		{
+			mergeQueryableScenario: threeTenantsScenario,
+			labelValuesTestCases: []labelValuesTestCase{
+				{
+					name:                "should return all label values for instance when no matchers are provided",
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a", "host2.team-b", "host2.team-c"},
+				},
+				{
+					name:                "should return all tenant team values for the __tenant_id__ label when no matchers are provided",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: []string{"team-a", "team-b", "team-c"},
+				},
+				// TODO(jdb): These tests are failing as matchers are not implemented. They should pass when the matchers are implemented
```

Comment:
I would add a field skipReason to the struct `labelValuesTestCase` and call T.Skip(skipReason) if `skipReason != ""`

Reply:
We later can remove once we have an implementation

Reply:
`T.Skip()` sounds like a great idea will definitely do that.